### PR TITLE
cmd/server: use redis=~5.0

### PR DESCRIPTION
**Pull request by bobheadxi**
The redis release we depend on for this image got pulled: https://pkgs.alpinelinux.org/packages?name=redis&branch=v3.12
[This is currently causing persistent build failures in main.](https://sourcegraph.slack.com/archives/C02FLQDD3TQ/p1633397543124500)

This PR makes the constraint looser so that we can pull the dependency at least, which comes with the unfortunate tradeoff that we cannot pin the version, but we also found the following:

> Unfortunately, Alpine-Linux Package Management drops older packages when there are newer versions available. This makes it hard to use Alpine Linux with docker since you want a reproducible image with exact versions.

https://superuser.com/questions/1174021/why-does-alpine-apk-report-unsatisfiable-constraints-when-installing-an-older/1486407#1486407

> You should avoid being explicit on package versions unless controlling your own package mirror and package builds, rather allow the package system to handle it for the Alpine release in use.

https://superuser.com/questions/1174021/why-does-alpine-apk-report-unsatisfiable-constraints-when-installing-an-older

Perhaps we shouldn't be pinning any of these dependencies 🤔

### CODE REVIEW

**Comment by bobheadxi** [comment]
Test build: https://buildkite.com/sourcegraph/sourcegraph/builds/110499#bbda0bb7-a708-4a6e-b4ae-9f481b8a713c

File: enterprise/dev/ci/internal/ci/runtype.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func computeRunType(tag, branch string) RunType {
 	case strings.HasPrefix(branch, "docker-images-patch/"):
 		return ImagePatch
 	case strings.HasPrefix(branch, "docker-images-patch-notest/"):
-		return ImagePatch
+		return ImagePatchNoTest
```
Comment (bobheadxi) [inline enterprise/dev/ci/internal/ci/runtype.go:49]: i wrote this bug here, fixed to get a test build going

**Comment by bobheadxi** [comment]
updated with the ~5.0 constraint: https://buildkite.com/sourcegraph/sourcegraph/builds/110503#6dcafbb5-e6e9-4ca9-a1b5-37f94aeba1ad

**Review by chrismwendt — APPROVED** [review]

**Comment by chrismwendt** [comment]
We ran into this before and ended up relaxing the constraints back then, too. There is a risk of non-reproducibility in relaxing these too much, but `=~5.0` (any patch version) seems very safe. Even `=~5` (any minor version) would probably be fine in terms of compatibility.

**Comment by bobheadxi** [comment]
However, it seems that if a `5.1` is introduced, the build will break again when the old version is pulled even if we keep the constraint for compatability hopes

**Comment by chrismwendt** [comment]
Yeah, so relaxing the constraint only reduces the frequency at which our builds break due to Alpine puling packages we depend on. If the frequency is low enough, it might be tolerable. If not, we can opt for one of the other solutions from StackOverflow (switch to some other distro like Ubuntu or set up an Alpine package mirror).

**Pull request merged**